llmfoundry/callbacks/async_eval_callback.py

```diff
@@ -0,0 +1,375 @@
+
+        cfg = self.current_run.submitted_config
+        default_compute = {
+            'gpus': 8,
+            'cluster': self.current_run.cluster,
+        }
+
+        run_name = get_run_name(self.current_run.name, str(current_interval))
+
+        params = get_eval_parameters(
+            parameters=self.training_config,
+            checkpoint=checkpoint,
+            training_run_name=self.current_run.name,
+        )
+        params['run_name'] = run_name
+
+        integrations = cfg.integrations
+        found_llm_foundry, installation_path = False, 'llm-foundry'
+        for i in integrations:
+            if i['integration_type'] != 'git_repo':
+                continue
+
+            if not i['git_repo'].endswith('llm-foundry'):  # detects forks
+                continue
+
+            found_llm_foundry = True
+            if i.get('path'):
+                installation_path = i['path']
+
+        if not found_llm_foundry:
+            from llmfoundry import __version__ as latest_foundry_version
+
+            # If github integration is not found, foundry is likely installed
+            # through the run command. In this case, we'll add the integration
+            # so the eval run will still work. However, it could cause unexpected
+            # behaviors because its not using custom repos or branches specified
+            # in the training run. For this reason, we'll log a warning
+            version = f'v{latest_foundry_version}'
+            log.warning(
+                'No github integration found for llm-foundry. Adding installation '
+                + f'to eval run for latest foundry release ({version}). ' +
+                'To use a fork, custom branch, or custom version, configure ' +
+                'llm-foundry installation through a github integration')
+            integrations.append({
+                'integration_type': 'git_repo',
+                'git_repo': 'mosaicml/llm-foundry',
+                'git_branch': version,
+                'pip_install': '-e .[gpu]',
+                'ssh_clone': False,
+            })
+
+        # This will record the timestamp and make it available for grouping
+        # and plotting in wandb
+        metadata = cfg.metadata
+        metadata['eval_timestamp'] = current_interval.value
+        metadata['eval_timestamp_unit'] = current_interval.unit.value
+
+        # TODO: This just runs an eval run, but we also want to attach the
+        # deployment, which would require a hf conversion and parametrizing the
+        # dependent_deployment in the run config
+        command = f'cd {installation_path}/scripts \n composer eval/eval.py $PARAMETERS'
+        run_config = RunConfig(
+            name=run_name,
+            image=self.current_run.image,
+            compute=self.compute or default_compute,
+            command=command,
+            integrations=integrations,
+            env_variables=cfg.env_variables,
+            metadata=cfg.metadata,
```

Comment:
In order for the metadata to make it to wandb, you'll need to copy https://github.com/mosaicml/llm-foundry/blob/5fdcc43bb31beccbd65bc0f18c5b12b7940445be/scripts/train/train.py#L469-L475 and https://github.com/mosaicml/llm-foundry/blob/5fdcc43bb31beccbd65bc0f18c5b12b7940445be/scripts/train/train.py#L613 from `train.py` to `eval.py`